pkg/provider/azure.go

```diff
@@ -1318,6 +1332,32 @@ func (az *Cloud) updateNodeCaches(prevNode, newNode *v1.Node) {
 	}
 }
 
+// updateNodeTaint updates node out-of-service taint
+func (az *Cloud) updateNodeTaint(node *v1.Node) {
+	if node == nil {
+		return
+	}
+	if isNodeReady(node) {
+		if az.KubeClient != nil {
+			if err := cloudnodeutil.RemoveTaintOffNode(az.KubeClient, node.Name, node, nodeOutOfServiceTaint); err != nil {
+				klog.Errorf("failed to remove taint %s from the node %s", v1.TaintNodeOutOfService, node.Name)
+			}
+		}
+	} else {
+		// node shutdown taint is added when cloud provider determines instance is shutdown
+		if !taints.TaintExists(node.Spec.Taints, nodeOutOfServiceTaint) &&
```

Comment:
when node is not ready, we cannot tell whether node is really shutdown or not

Reply:
Should we handle this scenario? I think we should use out-of-service taint carefully since it will force delete pod and detach volume.